src/libraries/System.Private.Xml/src/System/Xml/Xsl/XsltOld/SequentialOutput.cs

```diff
@@ -607,12 +606,8 @@ private void WriteHtmlUri(string value)
                     default:
                         if (127 < ch)
                         {
-                            if (_utf8Encoding == null)
-                            {
-                                _utf8Encoding = Encoding.UTF8;
-                                _byteBuffer = new byte[_utf8Encoding.GetMaxByteCount(1)];
-                            }
-                            int bytes = _utf8Encoding.GetBytes(value, i - 1, 1, _byteBuffer!, 0);
+                            _byteBuffer = new byte[Encoding.UTF8.GetMaxByteCount(1)];
```

Comment:
Isn't this change going to potentially cause a new buffer to be allocated for every non-ASCII char?

Reply:
Seems like a place for a stackalloc instead, no?

Reply:
I am thinking of using https://docs.microsoft.com/en-us/dotnet/api/system.text.encoding.getbytes?view=net-6.0#system-text-encoding-getbytes(system-char*-system-int32-system-byte*-system-int32) below.

Reply:
Did the old code work in multi-threading scenarios? I agree with @AaronRobinsonMSFT if we can use stackalloc and no need to call `Encoding.UTF8.GetMaxByteCount(1)` as it will always return a constant value.

Reply:
> Isn't this change going to potentially cause a new buffer to be allocated for every non-ASCII char?

Yes, good catch. I should have checked `_byteBuffer` - not `_utf8Encoding` - for null.

Reply:
> no need to call `Encoding.UTF8.GetMaxByteCount(1)` as it will always return a constant value.

Conveniently, due to the other changes made in this PR, this is a now burned in to the codegen as the constant value __6__. :)

Reply:
> Yes, good catch. I should have checked _byteBuffer - not _utf8Encoding - for null.

any concern in the multi-threading scenario if using the same _byteBuffer?

Reply:
This should ideally be changed to use `stackalloc` and `Rune`; avoid `Encoding.UTF8` entirely. But since this is a legacy code path I don't think I have a good way of testing this code path short of using reflection in a sample console app.

Given that this is legacy, might be best just to undo the change entirely.

Reply:
(This file has been reverted.)